### PR TITLE
[18.0][MIG] account_financial_report: Migration to 18.0

### PR DESCRIPTION
**Pull request by chaule97**
- OCA Port: https://github.com/OCA/account-financial-reporting/pull/1233
- I have renamed [account name](https://github.com/chaule97/account-financial-reporting/blob/befa9ceabbc8379c92dd03f4cf206ebcc11604c1/account_financial_report/tests/test_trial_balance.py#L77) "301" to "201" in `TestTrialBalanceReport` based on [SQL logic](https://github.com/odoo/odoo/blob/b20d26ca523a9971c05789256901b10bf2b488de/addons/account/models/account_account.py#L456) matching account codes to group prefixes.
- I have updated the test case test_05_all_accounts_loaded to only retrieve accounts with numeric codes, because there is [account](https://github.com/odoo/odoo/blob/b20d26ca523a9971c05789256901b10bf2b488de/addons/account/tests/common.py#L167) with string codes.

### CODE REVIEW

File: account_financial_report/report/templates/layouts.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    <template id="account_financial_report.internal_layout">
+        <div class="article o_account_financial_reports_page">
+            <t t-call-assets="web.assets_backend" />
+            <t t-out="0" />
+        </div>
+        <div class="footer">
+            <div class="row">
+                <div class="col-6 custom_footer">
+                    <span
+                        t-esc="context_timestamp(datetime.datetime.now()).strftime('%Y-%m-%d %H:%M')"
```
Comment (henrybackman) [inline account_financial_report/report/templates/layouts.xml]: `t-esc` should be replaced with `t-out`
[https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0)

File: account_financial_report/wizard/general_ledger_wizard_view.xml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        <field name="view_mode">form</field>
+        <field name="view_id" ref="general_ledger_wizard" />
+        <field
+            name="context"
+            eval="{
+                'default_receivable_accounts_only':1,
+                'default_payable_accounts_only':1,
+            }"
+        />
+        <field name="groups_id" eval="[(4, ref('account.group_account_manager'))]" />
```
Comment (henrybackman) [inline account_financial_report/wizard/general_ledger_wizard_view.xml:160]: `groups_id` should be replaced with `groups=` in the element
[https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0)

File: account_financial_report/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Author: Damien Crier
+# Author: Julien Coux
+# Copyright 2016 Camptocamp SA
+# Copyright 2020 ForgeFlow S.L. (https://www.forgeflow.com)
+# Copyright 2021 Tecnativa - Jo??o Marques
```
Comment (henrybackman) [inline account_financial_report/__manifest__.py]:
```suggestion
# Copyright 2021 Tecnativa - João Marques
```

File: account_financial_report/models/account.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# ?? 2011 Guewen Baconnier (Camptocamp)
```
Comment (henrybackman) [inline account_financial_report/models/account.py]:
```suggestion
# © 2011 Guewen Baconnier (Camptocamp)
```

Comment (henrybackman) [reply account_financial_report/models/account.py]: Check other files as well for similar issues

File: account_financial_report/report/aged_partner_balance.py [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+        elif today <= due_date + timedelta(days=90):
+            ag_pb_data[acc_id]["90_days"] += residual
+            ag_pb_data[acc_id][prt_id]["90_days"] += residual
+        elif today <= due_date + timedelta(days=120):
+            ag_pb_data[acc_id]["120_days"] += residual
+            ag_pb_data[acc_id][prt_id]["120_days"] += residual
+        else:
+            ag_pb_data[acc_id]["older"] += residual
+            ag_pb_data[acc_id][prt_id]["older"] += residual
+        if due_date:
```
Comment (henrybackman) [inline account_financial_report/report/aged_partner_balance.py]: Is it possible to not have `due_date`? It is set on line 59 `due_date = today`

File: account_financial_report/wizard/general_ledger_wizard_view.xml (hunk 2 above)
Comment (chaule97) [reply account_financial_report/wizard/general_ledger_wizard_view.xml:161]: Excuse me, I don't know where to put `groups=`?

**Comment by chaule97** [comment]
Hi @henrybackman, I have optimized the query, can you check it again?

**Review by henrybackman — APPROVED** [review]

**Review by salvorapi — APPROVED** [review]

**Comment by salvorapi** [comment]
LGTM

**Comment by victoralmau** [comment]
Please add https://github.com/OCA/account-financial-reporting/pull/1257 to commit history

**Comment by chaule97** [comment]
> Please add #1257 to commit history

thanks, I have added these commits

**Comment by victoralmau** [comment]
Please add https://github.com/OCA/account-financial-reporting/pull/1258 to commit history.

**Comment by victoralmau** [comment]
Please add https://github.com/OCA/account-financial-reporting/pull/1261 to commit history.

**Review by pedrobaeza — APPROVED** [review]
Tested on runboat

/ocabot merge nobump

File: account_financial_report/models/account_group.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+class AccountGroup(models.Model):
+    _inherit = "account.group"
+
+    group_child_ids = fields.One2many(
+        comodel_name="account.group", inverse_name="parent_id", string="Child Groups"
+    )
+    level = fields.Integer(compute="_compute_level", recursive=True)
+    account_ids = fields.One2many(
+        comodel_name="account.account", inverse_name="group_id", string="Accounts"
```
Comment (alexis-via) [inline account_financial_report/models/account_group.py:15]: We have a problem here: group_id on account.account was a stored field in v17, but it's NOT a stored field in v18. It seems to me that it is the cause of the message "2025-02-18 09:47:17,068 10499 ERROR o8_test1 odoo.osv.expression: Non-stored field account.account.group_id cannot be searched." ; you can see this message when you create a new company with a CoA for example. It breaks the tests in this PR https://github.com/OCA/l10n-france/pull/620 where I port a module that depend on account_financial_reporting

**Pull request merged**